From b98a7fcd2190f9b6e660fb3ae5bea279cc931f9b Mon Sep 17 00:00:00 2001
From: Abishek V Ashok <abishekvashok@gmail.com>
Date: Sat, 12 Dec 2020 17:38:26 +0530
Subject: [PATCH 1/2] Fixes #98: Properly initialise head to false

This fix introduces a loop to make sure that the head is properly
initialised so that random block characters don't appear on screen when
resizing.

Fixes #98
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>
(cherry picked from commit 7ab2d95e51e248336fa8fe9dda69c622dffd2126)
---
 cmatrix.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cmatrix.c b/cmatrix.c
index 90284cb..2fba9f4 100644
--- a/cmatrix.c
+++ b/cmatrix.c
@@ -181,6 +181,14 @@ void var_init() {
         matrix[i] = matrix[i - 1] + COLS;
     }
 
+    for (i = 0; i < LINES; ++i)
+    {
+        for (j = 0; j < COLS; ++j)
+        {
+            matrix[i][j].is_head = false;
+        }
+    }
+
     if (length != NULL) {
         free(length);
     }

From c44ee6d59e18cabf02b8e48722e77f442b928412 Mon Sep 17 00:00:00 2001
From: Abishek V Ashok <abishekvashok@gmail.com>
Date: Mon, 14 Dec 2020 12:01:07 +0530
Subject: [PATCH 2/2] Fixes #98: Don't try to print non-printable chars

This lead to unintended consequences as random block of characters
appeared on resizing when we tried to pass non printable -1 to addch.
Introduces a simple if statement to fix it.

Fixes #98
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>
(cherry picked from commit 6d620808c85f94db4806dd987429ae300bc1f844)
---
 cmatrix.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/cmatrix.c b/cmatrix.c
index 2fba9f4..efcd04b 100644
--- a/cmatrix.c
+++ b/cmatrix.c
@@ -181,14 +181,6 @@ void var_init() {
         matrix[i] = matrix[i - 1] + COLS;
     }
 
-    for (i = 0; i < LINES; ++i)
-    {
-        for (j = 0; j < COLS; ++j)
-        {
-            matrix[i][j].is_head = false;
-        }
-    }
-
     if (length != NULL) {
         free(length);
     }
@@ -689,6 +681,8 @@ if (console) {
                         } else {
                             addch('&');
                         }
+                    } else if (matrix[i][j].val == -1) {
+                        addch(' ');
                     } else {
                         addch(matrix[i][j].val);
                     }
